test/DerivativeTest.jl

```diff
@@ -133,4 +134,12 @@ end
     end
 end
 
+@testset "non-standard numbers" begin
+    furlong = Furlongs.Furlong{2}(1.0)
+
+    f(x) = exp(x) + 4*sin(x)*oneunit(x)
+
+    @test ForwardDiff.derivative(f, furlong) == exp(furlong) + 4*cos(furlong)
```

Comment:
Would any of `gradient` etc fail for Furlongs? If construct_seeds errors, it should also be possible to provoke errors in user-facing functions, and testing the latter seems much more relevant and future-proof than testing an internal function.